### PR TITLE
feat: use GKE module and defaults-gcp.yaml config

### PR DESCRIPTION
**Pull request by verbotenj**
Closes: #105 

Example of one node pool from the config

```
 # module.gke.google_container_node_pool.pools["co-mem-x86-az1"] will be created
  + resource "google_container_node_pool" "pools" {
      + cluster                     = "dmtr-cluster"
      + id                          = (known after apply)
      + initial_node_count          = 0
      + instance_group_urls         = (known after apply)
      + location                    = "us-east4"
      + managed_instance_group_urls = (known after apply)
      + max_pods_per_node           = (known after apply)
      + name                        = "co-mem-x86-az1"
      + name_prefix                 = (known after apply)
      + node_count                  = (known after apply)
      + node_locations              = [
          + "us-west2-b",
        ]
      + operation                   = (known after apply)
      + project                     = "blinklabs-demeter"
      + version                     = (known after apply)

      + autoscaling {
          + location_policy = (known after apply)
          + max_node_count  = 1
          + min_node_count  = 0
        }

      + management {
          + auto_repair  = true
          + auto_upgrade = true
        }

      + node_config {
          + disk_size_gb      = 100
          + disk_type         = "pd-ssd"
          + effective_taints  = (known after apply)
          + guest_accelerator = (known after apply)
          + image_type        = "COS_CONTAINERD"
          + labels            = {
              + "cluster_name"                  = "dmtr-cluster"
              + "demeter.run/availability-sla"  = "consistent"
              + "demeter.run/availability-zone" = "az1"
              + "demeter.run/compute-arch"      = "x86"
              + "demeter.run/compute-profile"   = "mem-intensive"
              + "node_pool"                     = "co-mem-x86-az1"
            }
          + local_ssd_count   = 0
          + logging_variant   = "DEFAULT"
          + machine_type      = "n4-highmem-8"
          + metadata          = {
              + "cluster_name"             = "dmtr-cluster"
              + "disable-legacy-endpoints" = "true"
              + "node_pool"                = "co-mem-x86-az1"
            }
          + min_cpu_platform  = (known after apply)
          + oauth_scopes      = [
              + "https://www.googleapis.com/auth/logging.write",
              + "https://www.googleapis.com/auth/monitoring",
            ]
          + preemptible       = false
          + service_account   = "terraform-runner@blinklabs-demeter.iam.gserviceaccount.com"
          + spot              = false
          + tags              = [
              + "gke-dmtr-cluster",
              + "gke-dmtr-cluster-co-mem-x86-az1",
            ]
            # (1 unchanged attribute hidden)

          + shielded_instance_config {
              + enable_integrity_monitoring = true
              + enable_secure_boot          = false
            }

          + taint {
              + effect = "NO_SCHEDULE"
              + key    = "demeter.run/availability-sla"
              + value  = "consistent"
            }
          + taint {
              + effect = "NO_SCHEDULE"
              + key    = "demeter.run/compute-profile"
              + value  = "mem-intensive"
            }
          + taint {
              + effect = "NO_SCHEDULE"
              + key    = "demeter.run/compute-arch"
              + value  = "x86"
            }

          + workload_metadata_config {
              + mode = "GKE_METADATA"
            }
        }

      + timeouts {
          + create = "45m"
          + delete = "45m"
          + update = "45m"
        }

      + upgrade_settings {
          + max_surge       = 1
          + max_unavailable = 0
          + strategy        = "SURGE"
        }
    }
    
   ```

### CODE REVIEW

File: common/defaults-gcp.yaml [hunk 1]
```diff
@@ -1,3 +1,230 @@
-# TODO: add GCP defaults
-region: us-east4
+cloud_provider: gcp
+cluster_name: dmtr-cluster
+region: us-west-2
```
Comment (wolf31o2) [inline common/defaults-gcp.yaml:3]: This can't be correct for GCP.

File: common/defaults-gcp.yaml [hunk 2]
```diff
@@ -1,3 +1,230 @@
...
+azs:
+  - us-west2-b
+  - us-west2-c
```
Comment (wolf31o2) [inline common/defaults-gcp.yaml:6]: These should be corrected for GCP.

File: common/defaults-gcp.yaml [hunk 3]
```diff
@@ -1,3 +1,230 @@
...
+        value: "consistent"
+        effect: NO_SCHEDULE
+      - key: demeter.run/compute-profile
+        value: "admin"
+        effect: NO_SCHEDULE
+      - key: demeter.run/compute-arch
+        value: "x86"
+        effect: NO_SCHEDULE
+    availability_zones:
+      - us-west2-b
```
Comment (wolf31o2) [inline common/defaults-gcp.yaml:34]: This should be corrected for GCP.

**Review by wolf31o2 — CHANGES_REQUESTED** [review]

File: common/defaults-gcp.yaml (hunk 1 above)
Comment (verbotenj) [reply common/defaults-gcp.yaml:3]: Good catch my config.yaml has those defined and it overrides it

File: common/defaults-gcp.yaml (hunk 2 above)
Comment (verbotenj) [reply common/defaults-gcp.yaml:6]:
```
gcloud compute zones list | grep us-west
us-west1-b                 us-west1                 UP
us-west1-c                 us-west1                 UP
us-west1-a                 us-west1                 UP
us-west2-a                 us-west2                 UP
us-west2-b                 us-west2                 UP
us-west2-c                 us-west2                 UP
```

File: stage0/gcp-terraform/main.tf [hunk 4]
```diff
@@ -52,75 +52,87 @@ resource "google_compute_subnetwork" "default" {
...
 }
 
-resource "google_container_cluster" "default" {
-  name = "dmtr-cluster"
+module "gke" {
+  source                     = "terraform-google-modules/kubernetes-engine/google"
+  project_id                 = local.project_id
+  name                       = local.name
+  region                     = local.region
+  network                    = google_compute_network.default.name
```
Comment (verbotenj) [inline stage0/gcp-terraform/main.tf:75]: I forgot to use `zones = local.azs`

File: stage0/gcp-terraform/main.tf [hunk 5]
```diff
@@ -52,75 +52,87 @@ resource "google_compute_subnetwork" "default" {
...
+  network_policy             = false
+  horizontal_pod_autoscaling = true
+  filestore_csi_driver       = false
+  dns_cache                  = false
+  deletion_protection        = false
+
+  node_pools = [for np in local.node_vars : {
+    name               = np.name
+    machine_type       = np.instance_type
+    min_count          = np.min_size
```
Comment (verbotenj) [inline stage0/gcp-terraform/main.tf:89]: I forgot `node_locations     = np.availability_zones` and it accept ""

**Review by wolf31o2 — APPROVED** [review]

**Pull request merged**